heartbeat/monitors/active/dialchain/tls.go

```diff
@@ -35,24 +36,36 @@ import (
 //        "rtt": { "handshake": { "us": ... }}
 //    }
 //  }
-func TLSLayer(cfg *transport.TLSConfig, to time.Duration) Layer {
+func TLSLayer(cfg *transport.TLSConfig, to time.Duration, meta *transport.DialerMeta) Layer {
 	return func(event common.MapStr, next transport.Dialer) (transport.Dialer, error) {
 		var timer timer
 
 		// Wrap next dialer so to start the timer when 'next' returns.
 		// This gets us the timestamp for when the TLS layer will start the handshake.
 		next = startTimerAfterDial(&timer, next)
 
-		dialer, err := transport.TLSDialer(next, cfg, to)
+		dialer, err := transport.TLSMetaDialer(next, cfg, to, meta)
 		if err != nil {
 			return nil, err
 		}
 
 		return afterDial(dialer, func(conn net.Conn) (net.Conn, error) {
 			// TODO: extract TLS connection parameters from connection object.
-
 			timer.stop()
 			event.Put("tls.rtt.handshake", look.RTT(timer.duration()))
+
+			connState := meta.TLS.ConnectionState()
+			certs := make([]common.MapStr, 0, len(connState.VerifiedChains))
+			for chainIndex, chain := range connState.VerifiedChains {
+				for certIndex, cert := range chain {
+					j := tls.CertToMap(cert, true)
+					j["chain_index"] = chainIndex
+					j["cert_index"] = certIndex
+					certs = append(certs, j)
+				}
+			}
+			event.Put("tls.certificates", certs)
```

Comment:
I'm a bit concerned that certificates here is going to be a nested object in Elasticsearch it seems.

What are the expected queries the user will run against these fields?